visualization/app/codeCharta/ui/artificialIntelligence/artificialIntelligence.component.ts

```diff
@@ -7,78 +7,286 @@ import { CustomConfigHelper } from "../../util/customConfigHelper"
-import { getMedian, pushSorted } from "../../util/nodeDecorator"
-import { NodeType } from "../../codeCharta.model"
+import { pushSorted } from "../../util/nodeDecorator"
+import { ColorRange, NodeType, State } from "../../codeCharta.model"
 import { hierarchy } from "d3-hierarchy"
 import { getVisibleFileStates } from "../../model/files/files.helper"
-
-type OutlierConfig = {
-	threshold: number
-}
+import { clusterMetricThresholds } from "./artificialIntelligence.clusterMetricThresholds"
+import { defaultMapColors } from "../../state/store/appSettings/mapColors/mapColors.actions"
+import { setState } from "../../state/store/state.actions"
+import { setColorRange } from "../../state/store/dynamicSettings/colorRange/colorRange.actions"
+import { setMargin } from "../../state/store/dynamicSettings/margin/margin.actions"
+import { ThreeOrbitControlsService } from "../codeMap/threeViewer/threeOrbitControlsService"
+import { ThreeCameraService } from "../codeMap/threeViewer/threeCameraService"
+import { setCamera } from "../../state/store/appSettings/camera/camera.actions"
+import { setCameraTarget } from "../../state/store/appSettings/cameraTarget/cameraTarget.actions"
+import { Vector3 } from "three"
 
 interface MetricValues {
 	[metric: string]: number[]
 }
 
+interface MetricAssessmentResults {
+	suspiciousMetrics: Map<string, ColorRange>
+	unsuspiciousMetrics: string[]
+	outliersThresholds: Map<string, number>
+}
+
+interface MetricSuggestionParameters {
+	metric: string
+	from: number
+	to: number
+	generalCustomConfigId: string
+	outlierCustomConfigId?: string
+}
+
 export class ArtificialIntelligenceController implements FilesSelectionSubscriber {
-	private metricsOutlierConfig: Map<string, OutlierConfig> = new Map([
-		["mcc", { threshold: 15 }],
-		["loc", { threshold: 1000 }],
-		["rloc", { threshold: 1000 }],
-		["cognitive_complexity", { threshold: 10 }],
-		["code_smell", { threshold: 25 }],
-		["bug", { threshold: 10 }],
-		["functions", { threshold: 20 }],
-		["statements", { threshold: 150 }]
-	])
+	private _viewModel: {
+		isFeatureApplicable: boolean
+		suspiciousMetricSuggestionLinks: MetricSuggestionParameters[]
+		unsuspiciousMetrics: string[]
+		riskProfile: {
+			lowRisk: number
+			moderateRisk: number
+			highRisk: number
+			veryHighRisk: number
+		}
+	} = {
+		isFeatureApplicable: false,
+		suspiciousMetricSuggestionLinks: [],
+		unsuspiciousMetrics: [],
+		riskProfile: {
+			lowRisk: 0,
+			moderateRisk: 0,
+			highRisk: 0,
+			veryHighRisk: 0
+		}
+	}
 
 	/* @ngInject */
-	constructor(private $rootScope: IRootScopeService, private storeService: StoreService) {
+	constructor(
+		private $rootScope: IRootScopeService,
+		private storeService: StoreService,
+		private threeOrbitControlsService: ThreeOrbitControlsService,
+		private threeCameraService: ThreeCameraService
+	) {
 		FilesService.subscribe(this.$rootScope, this)
 	}
 
+	applyCustomConfig(configId: string) {
```

Comment:
This is a Config from `customConfig.component.ts`. I thought to not make this component dependent on the customConfigs component. But this doesn't make sense, because the algo adds Custom Configs as its results.
So we can reuse the original method.